Add customisation

Andy-Bell opened the pull request:
Updates CORSMiddleware with two new functions
needed to create new instances of the middleware
allowing customisation of the headers, with a
"default" function providing an instance of the
middleware that functions the same as in the
0.1.0 version, and a "new" function that allows
the end user to use differing values as they
find the need to in their use case.

Andy-Bell committed:
Updates CORSMiddleware with two new functions

needed to create new instances of the middleware
allowing customisation of the headers, with a
"default" function providing an instance of the
middleware that functions the same as in the
0.1.0 version, and a "new" function that allows
the end user to use differing values as they
find the need to in their use case.

Andy-Bell committed:
Bump version number

Andy-Bell committed:
Update documentation and README file.

Adds in basic documentation on useage of the
middleware to the lib.rs file.

Updates the README to show the changes in this
version (builder in place, basic documentation)
and to also add minimum version expected (due
to colleague having build issues on earlier rust
version.

dcorking commented on src/lib.rs:
```diff
@@ -15,25 +20,93 @@ use hyper::header::{
+}
+
+impl CORSMiddleware {
+    pub fn new(methods: Vec<Method>, origin: Option<String>, max_age: u32) -> CORSMiddleware {
```
Can I suggest a brief docstring for #new(), possible with an example call? (either here or in the main struct docstring)

possibly you can automate copying the main docstring into the README as a git commit hook, so the README doesn't get out of sync

dcorking reviewed (approved):
really clear and easy to follow tests

🚂

Andy-Bell committed:
Add further documentation comments around the new

and default public methods to explain the main
differences between the methods.

Andy-Bell committed:
Adds example for use of CORSMiddleware::new()

Refactors the CORSMiddleware::default() method
to make it make use of the CORSMiddleware::new()
internally so that if any future changes are made
to how the middleware builds, it will still be
reflected in how the default version is built
(as the default is just supposed to be a series
of pre-defined values used and not actually differ
on the building)

dcorking reviewed (approved):
🚂
